packages/admin-panel/src/VizBuilderApp/api/fetchError.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ *
+ */
+export default class FetchError extends Error {
+  constructor(message, code, ...params) {
+    // Pass remaining arguments (including vendor specific ones) to parent constructor
+    super([message, ...params]);
```

Comment:
I see that this is also used in `lesmis`, does this work? I would expect the following syntax:

```suggestion
    super(message, ...params);
```

Reply:
Yep this is correct the api is that the constructor takes an array!

Reply:
This is the JS Error class right? I checked the docs and also tried this locally but couldn't get it working with an array...

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error - "ES6 Custom Error Class"